packages/cantata/use_termux_path.patch

```diff
@@ -0,0 +1,109 @@
+     }
+     return instance()->mDataDirs;
+   } else if ( qstrncmp( "config", resource, 6 ) == 0 ) {
+     if ( instance()->mConfigDirs.isEmpty() ) {
+-      instance()->mConfigDirs = instance()->systemPathList( "XDG_CONFIG_DIRS", "/etc/xdg" );
++      instance()->mConfigDirs = instance()->systemPathList( "XDG_CONFIG_DIRS", "@TERMUX_PREFIX@/etc/xdg" );
+     }
+     return instance()->mConfigDirs;
+   }
+diff --git a/gui/settings.cpp b/gui/settings.cpp
+index 5c77961..82906ee 100644
+--- a/gui/settings.cpp
++++ b/gui/settings.cpp
+@@ -39,7 +39,7 @@ struct MpdDefaults
+ {
+     MpdDefaults()
+         : host("localhost")
+-        , dir("/var/lib/mpd/music/")
++        , dir("@TERMUX_PREFIX@/var/lib/mpd/music/")
+         , port(6600) {
+     }
+ 
+@@ -58,7 +58,7 @@ struct MpdDefaults
+     };
+ 
+     void read() {
+-        QFile f("/etc/mpd.conf");
++        QFile f("@TERMUX_PREFIX@/etc/mpd.conf");
+ 
+         if (f.open(QIODevice::ReadOnly|QIODevice::Text)) {
+             int details=0;
+diff --git a/playlists/cantata-dynamic b/playlists/cantata-dynamic
+index 110f65a..800bb46 100755
+--- a/playlists/cantata-dynamic
++++ b/playlists/cantata-dynamic
+@@ -1071,8 +1071,8 @@ sub stop() {
+ # #####################################
+ # SERVER MODE
+ # #####################################
+-$filesDir="/var/lib/mpd/dynamic";
+-$pidFile="/var/run/cantata-dynamic/pid";
++$filesDir="@TERMUX_PREFIX@/var/lib/mpd/dynamic";
++$pidFile="@TERMUX_PREFIX@/var/run/cantata-dynamic/pid";
+ 
+ sub encodeString() {
+     my $str=shift;
+@@ -1098,10 +1098,10 @@ sub decodeString() {
+ sub loadConfig() {
+     my $config=shift;
+     if (!$config || ($config=~ m/^(default)/)) {
+-        $config="/etc/cantata-dynamic.conf";
++        $config="@TERMUX_PREFIX@/etc/cantata-dynamic.conf";
+     }
+     open(my $fileHandle, $config)  || die "ERROR: Failed to load config $config - $!\n";
+-    $activeFile="/var/run/cantata-dynamic/rules";
++    $activeFile="@TERMUX_PREFIX@/var/run/cantata-dynamic/rules";
+     if (tell($fileHandle) != -1) {
+         my @lines = <$fileHandle>; # Read into an array...
+         close($fileHandle);
+diff --git a/playlists/cantata-dynamic.service b/playlists/cantata-dynamic.service
+index 53ebc41..1a91512 100755
+--- a/playlists/cantata-dynamic.service
++++ b/playlists/cantata-dynamic.service
+@@ -7,8 +7,8 @@ After=mpd.service
+ User=mpd
+ Group=audio
+ Type=forking
+-ExecStart=/usr/share/cantata/scripts/cantata-dynamic server /etc/cantata-dynamic.conf
+-ExecStop=/usr/share/cantata/scripts/cantata-dynamic stopserver /etc/cantata-dynamic.conf
++ExecStart=/data/data/com.termux/files/usr/share/cantata/scripts/cantata-dynamic server /data/data/com.termux/files/usr/etc/cantata-dynamic.conf
++ExecStop=/data/data/com.termux/files/usr/share/cantata/scripts/cantata-dynamic stopserver /data/data/com.termux/files/usr/etc/cantata-dynamic.conf
```

Comment:
In a patch you can use `@TERMUX_PREFIX@` instead of `/data/data/com.termux/files/usr`.